app/models/content_item.rb

```diff
@@ -9,9 +10,27 @@ def self.create_or_update!(attributes, organisation)
     content_id = attributes.fetch(:content_id)
     content_item = self.find_or_create_by(content_id: content_id)
 
-    attributes = attributes.slice(*content_item.attributes.symbolize_keys.keys)
+    content_item.add_organisation(organisation)
+    content_item.add_taxonomies(attributes.fetch(:taxons))
 
+    attributes = content_item.existing_attributes(attributes)
     content_item.update!(attributes)
-    content_item.organisations << organisation unless content_item.organisations.include?(organisation)
+  end
+
+  def add_organisation(organisation)
+    self.organisations << organisation unless self.organisations.include?(organisation)
+  end
+
+  def add_taxonomies(taxon_ids)
+    taxon_ids.each do |taxon_id|
+      taxon = Taxonomy.find_by(content_id: taxon_id)
+      if taxon && !self.taxonomies.include?(taxon)
```

Comment:
Addressed in 035359ae3b204361f19a453bfd7dbd2e7cb94653 and the unique index in 39188da65a135ec67f3f43cb754c28b4e696946c